### DISCUSSION TITLE
1.3.3.20 new compiler version

### DISCUSSION THREAD

**Discussion by mha1** [pascallanger/DIY-Multiprotocol-TX-Module#740] [General]
Hi Pascal,

can you please elaborate on this item listed in the release notes for 1.3.3.20. Is there anything to do to keep the home brew build system up to date?

Thanks,
Michael

**Comment by mha1** [comment]
anyone?

**Comment by benlye** [comment]
Hi Michael, if you want to use the new compiler version when compiling yourself you can switch to the 'devel' version of the MULTI boards package.

You need to use this URL for the additional boards option in the IDE:
`https://raw.githubusercontent.com/pascallanger/DIY-Multiprotocol-TX-Module-Boards/devel/source/package_multi_4in1_board_devel_index.json`

To avoid confusion, I would recommend uninstalling the original MULTI boards first as the devel ones have the same names and it can be confusing to have both installed (hard to know which one you have selected).

The new boards package takes a while to install so don't panic if it looks like it's not doing anything for a bit.

If all goes well and there are no issues reported we intend to switch the main boards package over to the new compiler soon.

**Reply by mha1** [reply]
Thank you very much for the quality response. According to the release notes 1.3.3.20 assets were already built with the new compiler version so I guess it's ok for production code. Was there a specific reason to change horses?

Thanks,
Michael

**Comment by benlye** [comment]
We're as confident as we can be that the new compiler is producing reliable binaries. 

We made the switch because the newer compiler builds fractionally smaller binary files and we're at the point where every byte matters on the STM32.

There are some more details in #538.

**Comment by mha1** [comment]
Thanks for the details. I second OpenUAS's statement but as it was made one year ago I seem to have missed the boat a long time ago. I appreciate your effort!
